assignments/array-methods.js

```diff
@@ -56,28 +56,66 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 let fullName = [];
+// fullName = runners.forEach((runner) => {
+//    return {'first_name':runner.first_name, 'last_name': runner.last_name};
+//    consle.log(runner.first_name)
+// });
+runners.forEach((runner) => {
+    fullName.push({"first_name": runner.first_name, "last_name": runner.last_name});
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps = runners.map((runner) => {
+    return runner.first_name.toUpperCase();
+});
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter((runner) => {
+   return runner.shirt_size === 'L';
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((total, runner) => {
+    return total += runner.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 // create a list of super donors so we can send them personalized thank yous. The list must be only of company names. A super donor donated 200 or more.
+let superDonors = runners.filter((runner) => {
+    return runner.donation >= 200;
+}).map((runner) => `Company name: ${runner.company_name}`);
+console.log(superDonors);
+1
+// Problem 2 // We've decided to only send out a specialized email thank you to super donors, so now we need a list of super donor objects with email addresses, company names and donation amounts(hint: for filtering)
+//first we are going to transform the entire runners array of objects into an array of objects with only email and company name properties
+let superDonorContact = runners.map((runner) => {
+    return {"email":runner.email, "company_name": runner.company_name, "donation": runner.donation};
+}).filter((runner) => runner.donation >= 200); // filtering the mapped list of email, comp names, and donations into a smaller list of super donors.
```

Comment:
For dot-chaining, you would hit enter right before `.map` and then again right before `.filter`.